docs/source/config/splitters/heka_framing.rst

```diff
@@ -0,0 +1,54 @@
+
+HekaFramingSplitter
+===================
+
+A HekaFramingSplitter is used to split streams of data that use Heka's built-
+in :ref:`stream_framing`, with a protocol buffers encoded message header
+supporting HMAC key authentication.
+
+A default configuration of the HekaFramingSplitter is automatically registered
+as an available splitter plugin as "HekaFramingSplitter", so it is only
+necessary to add an additional TOML section if you want to use an instance of
+the splitter with settings other than the default.
+
+Config:
+
+- signer:
+	Optional TOML subsection. Section name consists of a signer name, underscore,
+	and numeric version of the key.
+
+	- hmac_key (string):
+	    The hash key used to sign the message.
+
+- use_message_bytes (bool, optional):
+	The HekaFramingSplitter is almost always used in concert with an instance
+	of ProtobufDecoder, which expects the protocol buffer message data to be
+	available in the PipelinePack's MsgBytes attribute, so `use_message_bytes`
+	defaults to true.
+
+- skip_authentication (bool, optional):
+	Usually if a HekaFramingSplitter identifies an incorrectly signed message,
+	that message will be silently dropped. In some cases, however, such as
+	when loading a stream of protobuf encoded Heka messages from a file system
```

Comment:
Yeah, we haven't been writing signed messages to disk, but in the previous implementation the LogstreamerInput was explicitly ignoring the authentication. I put this in so it'd be possible to replicate the prior behaviour.

You're right re: the duplicate header decoding. It's technically possible to cache the header from a successful FindRecord call and reuse it in UnframeRecord, since those calls are made serially from within the same goroutine. We could make that an explicit part of the contract, so folks know it's safe to store struct attributes on their splitter btn the two calls. But signed messages are used so infrequently it didn't seem worth it. I'll happily update if you feel differently, though. :)